src/com/gfredericks/test/chuck/generators.clj

```diff
@@ -227,3 +227,17 @@
   any of those features."
   [regex]
   (regexes/gen-string-from-regex regex))
+
+(defn sub-map
+  "Given a concrete map, it'll randomly select keys
+   from it thus making it a subset of the given map.
+   Note: It can return empty maps as well.
+
+   Example:
+   (gen/sample (sub-map {:a 1 :b 2 :c 3}))
+   => ({} {:b 2} {:b 2, :c 3} {:a 1} ...)"
+  [m]
+  (gen/fmap (fn [[m ks]]
+              (select-keys m ks))
+            (gen/tuple (gen/return m)
```

Comment:
using `tuple` here to get `m` isn't necessary since you already have `m` available in the lexical scope